src/async-function.ts

```diff
@@ -192,29 +184,30 @@ export class State<Value> {
    * `error` or `resolvedValue` will remain as they were previously
    * until this promise resolves, and then they'll be updated to the new values.
    */
-  retry = async () => {
+  async retry() {
```

Comment:
imo, this should remain as a property who's value is a function -- if someone does `const { retry, isLoading, isError } = state`, they'll lose the `this` context.

This is obnoxious, and I think it's one of JS's biggest mistakes when it comes to classes. :-\

Reply:
> if someone does const `{ retry, isLoading, isError } = state`, they'll lose the this context.

This true of all class instance and object methods though. I don't think all methods should be written as fields, since that's wasteful, and hopefully consumers of objects know that they can't just tear off methods unless the method documents that it's bound to the instance.

For consistency sake, I personally would stick to plain methods, unless the method/function is specifically documented and designed to be forwarded to another invoker (like event listeners).

I'll of course make the change if you want though!

Reply:
I understand -- but I think bound methods for public API is a good default, ye <3

Reply:
I changed it back, but I have to say that the vast majority of class methods out there are simply methods and not bound instance fields, so they aren't safe to tear off.

That makes this "method" special, and JavaScript developers will still always need to know to call `o.someMethod.bind(o)` in most cases when they want to tear off methods. So I personally don't think it's a great default, because the default is already established by the language semantics, all of the built-in APIs, the DOM, and nearly all classes in library code out there.

One new place that does take special care to make bound functions is `Promise.withResolvers()`. This is a static method that returns a plain object though, not a class instance. I wonder if there's some style guidance in TC39 about that? Maybe `signalFunction()` should return a plain object too if it's intended to be used similarly?

My disagreement is pretty low temperature here though, and it's your code base!